### DISCUSSION TITLE
4/25 멘토링

### DISCUSSION THREAD

**Discussion by ms1011** [OmokNoonE/PPM-backend#3] [멘토님 미팅]
# 멘토 미팅

### 주제 선정

- Jira 형태의 프로젝트는 비교될 수밖에 없다
- 캘린더를 하게 되면 구글 캘린더를 만들기보단 연계되는 프로그램을 만드는 것이 더 나을 것 같다
- 타겟층이 중요하다. 누가 사용하는 것인지가 가장 중요하다
- 특별한 알람을 주는 기능?
- 관리자를 위한 일정관리 프로젝트 괜찮은 것 같다?
- 드롭다운은 불편하다 -> 편하게 기능을 개발해야한다
- 일정관리는 대시보드와 알림이 중요하다!!!
- 우리가 사용하는 프로젝트를 관리하는 프로젝트? or 학원다니는 사람들이 사용하는 프로젝트
- 달력은 오히려 프로젝트 관리 형태에서는 불편하다
- 5단계 설계를 기준으로 4단계도 가능하게, 각 단계별로 일정관리가 가능하게 하는것이 어떤지
- 관리자는 실무자와 소통을 직접 or 메일, 메신저를 통해 관리한다
- 실제로 우리가 걸렸던 일정을 팁으로 제공해주는 것도 좋다

### 실제 프로젝트 진행 될 경우

- 현업에서는 개발자는 아이디어를 잘 내지 않는다.
- 개발자가 기획을 주도하면 다른 시스템과 호환에 문제가 생기는 경우가 있어 넓은 범위에서 생각해야 한다
- 프로젝트가 시작되면, D-Day를 정하고, 역산을하여 일정을 조율한다
- 폭포수 형태로 개발 진행 (짧은 기간에 가장 성공하기 쉽다)
- 기획, 설계 단계에서 어느정도 정해지게 되면 계획은 크게 변경되지 않는다
- 정책은 요구사항 분석 단계에서 정해진다 (멘토님께서 직접 도움을 주실 수 있다)

## 다음 미팅까지 준비할 것

1. 요구사항 명세서 (같은 반 사람들에게 요구사항 분석하기)
2. 단어사전
3. 인터뷰 계획서 (질문 리스트)

**Comment by jlee38266** [comment]
폭포수 5단계 개발 방식을 채택하되, 필요 따라 특정 단계는 약소 또는 생략을 하는 방향도 나쁘지 않을 것 같습니다

**Reply by jlee38266** [reply]
프로젝트에 진행에 관련하여 가중치를 만들고 혹여나 계획 보다 일정이 여유롭게 종료 되는 경우 다른 테스트 부분에 가중치를 더 받는 것이 좋다는 피드백도 있었습니다.

**Comment by jihyeongeun** [comment]
1. 분석 및 화면 설계 약 7일
2. 개발 약 25일
3. 테스트 약 10일(길수록 좋음)
4. 배포기간 약 4일 정도로 일정을 잡으면 좋을 것 같다고 하셨었습니다.

5.1(수) 오전 9시 2차 미팅 예정입니다!

**Comment by CJC0512** [comment]
### 다음과 같은 추가적인 의견이 있습니다.
- 일정 등록 시, Input 값은 [업무, 시작일, 종료일]과 같이 최소한의 속성으로 간단하게 등록할 수 있도록 설계 (부가적인 속성은 추후 수정 가능하겠끔 유도.)
- Geek 스러운 기능으로 Tooltip이나 일정 지연과 같은 부분에서 재미를 줄 수 있는 요소 추가. (ex. 일정 지연 시, 일정에 불이 난다.)
- 일정 작성에 있어, 은어 혹은 약어를 사용할 수 있기에, Wiki 기능을 추가.

**Comment by jlee38266** [comment]
# 정리

 ## 주제
- 부트 캠프 학생들을 위한 일정 관리 시스템

 ## 선정 이유
- 부트 캠프 과정을 통해 프로젝트를 진행하면서 일정 관리의 어려움을 느끼고 이런 부분을 해소할 수 있는 일정 관리 시스템이 있으면 좋겠다라는 생각에서 출발

 ## 핵심기능
- 일정을 등록하면 한 눈에 확인할 수 있는 대시보드를 제공한다.
- 마감일이 다가오면 적절한 알림을 사용자에게 제공 한다.

**Comment by jlee38266** [comment]
# 정리 2  (2024/04/29)

 ##  변경 된 주제
- 프로젝트 매너저가 소규모 프로젝트에서 활용하기 좋은 일정 관리 시스템, 핵심적으로 제공 하는 기능은 크게 변동이 없으나, 타켓층을 좀 더 구체적으로 선정하여 적절한 기능을 개발하기 위해서 변경되었습니다.

 ## 선정 이유
-  PM의 입장에서 대규모 프로젝트 이외에 개인 또는 소규모 프로젝트를 관리할 때 일정 작성이 간단하지만 시각적으로 잘 보여질 수 있는    erp가 있다면 좋을 것 같다는 니즈를 파악

 ## 핵심기능
- 일정을 등록하면 한 눈에 확인할 수 있는 대시보드를 제공한다.
- 마감일이 다가오면 적절한 알림을 사용자에게 제공 한다.

**Comment by jlee38266** [comment]
## 선호도 조사
- google forms을 활용하여 일정 관리 툴에 대한 기능적 니즈 파악: https://forms.gle/3CUFXwQPSXpENviC8

**Comment by jlee38266** [comment]
# 회의 내용   (2024/04/30)

 ##  논의 사항
- 외부 관계자 연락처에 관한 정보를 저장 하는 테이블이 따로 필요한가 (만약 필요 없을 시, 공유 부분에 관한 테이블도 필요 없을 수 있음)
- 가중치 속성의 적용 방법에 대한 구체적인 가이드라인 필요
- 회원 가입 관련된 자세한 논의 필요 - 실제 가입 과정 (관리의 주체가 누구인가), 관리 되는 방식 또는 구현 되었으면 하는 방식
- 소프트웨어 아키텍쳐 점검
- 화면 설계서 함께 점검

**Reply by jlee38266** [reply]
- 일정을 계획 할 때 다수가 참여 하는가, 그리고 일정에 참여 하고 있는 인원이 개인의 필요에 의해서 일정에 관련 하여 수정 요청을 하는가 (개인이 그러한 일정 변화를 직접 일정에 반영할 수 있는가)
- 프로젝트에 참여 하고 있는 인원이 일정 관련된 erp의 접근 권한은 어디까지? 읽기만 가능, 읽기 쓰기 가능, 다른 인력 추가 가능 등등

**Reply by jlee38266** [reply]
- 실제 마감일을 지키지 못했을 때를 고려한 속성값 추가에 대한 논의 필요 - 연장일 개념이 존재는 하지만 실제로 프로젝트 진행 환경에서 연장 신청을 하지 않고 주어진 기간 보다 초과하여 작업을 마무리 했을 때를 어떻게 관리하는가

**Reply by jlee38266** [reply]
권한 부여에 관련 하여 사용자가 특정 권한을 사용할 수 있도록 (커스터마이스 가능) 해야 하는지 아니면 부여된 권한으로만 이용을 해야 하는지

**Reply by jlee38266** [reply]
한 회사에서만 활용한다는 가정 하에 진행할지, 여러 회사에서 사용한다는 가정 하에 진행할지.
회사코드 혹은 ID 필요여부 설정을 위해
